### ISSUE TITLE
Time sequence prediction still broken

### ISSUE THREAD

**Issue by plang85** [pytorch/examples#243]
Backed out #222 and it seems to work again.

**Comment by plang85** [comment]
Just saw that this is still being duscussed in a previously closed issue #215

**Comment by plang85** [comment]
In it's current state the example is broken, so I think this deserves to be an open issue, pls correct me if I'm wrong

**Comment by plang85** [comment]
nevermind me, all works as advertised if installed using conda pm

**Comment by vgoklani** [comment]
bump

**Comment by barnabytprowe** [comment]
Thanks for linking this issue @plang85 .

I had been having trouble with this example and had put some results on #215.  I had to reduce the learning rate `lr` significantly to achieve a "good" solution.

I ran three cases spanning the decade from `lr` = 0.01 up to `lr` = 0.10.

These are below (gzipped avi and stdout output from the code):
lr = 0.01:
[lr_0.01_epochs_300_predicts.avi.gz](https://github.com/pytorch/examples/files/1435747/lr_0.01_epochs_300_predicts.avi.gz)
[lr_0.01_epochs_300_predicts.stdout.txt.gz](https://github.com/pytorch/examples/files/1435748/lr_0.01_epochs_300_predicts.stdout.txt.gz)

lr = 0.03:
[lr_0.03_epochs_300_predicts.avi.gz](https://github.com/pytorch/examples/files/1435749/lr_0.03_epochs_300_predicts.avi.gz)
[lr_0.03_epochs_300_predicts.stdout.txt.gz](https://github.com/pytorch/examples/files/1435750/lr_0.03_epochs_300_predicts.stdout.txt.gz)

lr = 0.10:
[lr_0.10_epochs_300_predicts.avi.gz](https://github.com/pytorch/examples/files/1435751/lr_0.10_epochs_300_predicts.avi.gz)
[lr_0.10_epochs_300_predicts.stdout.txt.gz](https://github.com/pytorch/examples/files/1435752/lr_0.10_epochs_300_predicts.stdout.txt.gz)

The videos are 10 steps / epochs per second, 300 epochs / 30 seconds total. If anyone would rather have the videos in a .zip format let me know.

As you can see, only the `lr` = 0.01 case converges to a solution which repeats in the way illustrated in the example. Both others find a different stable minimum in the cost function, in the `lr` = 0.10 case with a lower final cost than the "better" case that repeats stably.

I think
- when training, better fits to the training data can often be models that don't repeat well
- the slower learning rate helps the `lr` = 0.1 model move stably to "good" repeating solution for this one example configuration of random initial weights
- results will probably vary significantly if we use a different random initialisation of the weights

So I don't know too much what to read into this. But it might be good to change the learning rate in the example to 0.01 and set the number of learning steps / epochs to 150. The main issue is the code then takes a while to run, over 2 hours on my (old) laptop.

**Comment by barnabytprowe** [comment]
I think this discussion is tangentially relevant: https://discuss.pytorch.org/t/lstm-time-sequence-generation/1916

As is this: https://discuss.pytorch.org/t/lstm-time-sequence-generation/1916

**Comment by drscotthawley** [comment]
@barnabytprowe thanks for pointing out the variability in the learning rate.  Looking at the history, it seems that @boathit and @soumith changed the learning rate to 0.8.  

@boathit or @soumith , can you comment on this?  Were you actually able to get viable output with a learning rate of 0.8? 
Because when I run this example as is, I get the following plot instead of the one displayed for this example: https://discuss.pytorch.org/uploads/default/original/2X/2/221e3389d4d569afff9d073c08b36a1b1e0604b6.jpg

**Comment by t-vi** [comment]
You don't want the nonlinearity at the output.
Add a linear layer or so, e.g. https://gist.github.com/t-vi/886eed2dda02319e5d111799eb2446be

**Comment by soumith** [comment]
fixed via #250
